src/configuration.ts

```diff
@@ -28,32 +28,29 @@ export function load(options: ConfigLoaderOptions = {}): Configuration {
   return fromPath(rootPath, options);
 }
 
-function getPackages(rootPath: string): [{ name: string; path: string }] | [] {
-  let packages = [];
-
-  if (fs.existsSync(path.join(rootPath, "package-lock.json"))) {
-    const result = execa.sync("npm", ["query", ".workspace"], { cwd: rootPath });
-    const workspaceQuery = JSON.parse(result.stdout);
-
-    packages = workspaceQuery.map((item: any) => ({ name: item.name, path: item.path }));
-  } else if (fs.existsSync(path.join(rootPath, "pnpm-lock.yaml"))) {
-    const result = execa.sync(`pnpm`, ["m", "ls", "--json", "--depth=-1"], { cwd: rootPath });
-    const workspaceJson = JSON.parse(result.stdout);
-
-    packages = workspaceJson
-      .filter((item: any) => item.name && item.path)
-      .map((item: any) => ({ name: item.name, path: item.path }));
-  } else if (fs.existsSync(path.join(rootPath, "yarn.lock"))) {
-    const result = execa.sync(`yarn`, ["--silent", "workspaces", "info", "--json"], { cwd: rootPath });
-    const workspaceMap = JSON.parse(result.stdout);
-
-    packages = Object.keys(workspaceMap).map(key => ({
-      name: key,
-      path: path.resolve(rootPath, workspaceMap[key].location),
-    }));
-  }
+interface PackageJson {
+  type: boolean
+  name: string;
+}
+
+interface Package {
+  dir: string;
+  relativeDir: string;
+  packageJson: PackageJson;
+}
+
+interface PackagesResult {
+  tool: {
+    type: 'pnpm' | 'yarn' | 'npm';
+  };
+  packages: Package[]
+  rootPackage: Package;
+}
+
+function getPackages(rootPath: string): { name: string; path: string }[] {
+  let { packages } = getPackagesSync(rootPath) as PackagesResult;
```

Comment:
I couldn't find types for manypkg, so I just made a rough minimal set, based on what I was seeing as output